packages/server/graphql/types/User.ts

```diff
@@ -421,19 +423,38 @@ const User = new GraphQLObjectType<any, GQLContext>({
       },
       resolve: async ({id: userId}, {reflectionId}, {dataLoader}) => {
         const retroReflection = await dataLoader.get('retroReflections').load(reflectionId)
-        if (!retroReflection) return []
+        if (!retroReflection) {
+          return standardError(new Error('Invalid reflection id'), {userId})
+        }
         const {meetingId} = retroReflection
         const meetingMemberId = toTeamMemberId(meetingId, userId)
-        const [viewerMeetingMember, reflectionGroups] = await Promise.all([
+        const r = await getRethink()
+        const [viewerMeetingMember, reflections] = await Promise.all([
           dataLoader.get('meetingMembers').load(meetingMemberId),
-          dataLoader.get('retroReflectionGroupsByMeetingId').load(meetingId)
+          r
+            .table('RetroReflection')
+            .getAll(meetingId, {index: 'meetingId'})
+            .filter({isActive: true})
+            .orderBy('createdAt')
+            .run()
         ])
         const {meetingId: viewerMeetingId} = viewerMeetingMember
         if (viewerMeetingId !== meetingId) {
-          standardError(new Error('Not on team'), {userId})
-          return []
+          return standardError(new Error('Not on team'), {userId})
         }
-        return reflectionGroups
+        const {groups} = groupReflections(reflections, AUTO_GROUPING_THRESHOLD)
```

Comment:
Hey @tianrunhe, it looks like you've played with the `groupReflection` function before. I'd love to get your help here.

The AC says:
> The heuristic above should attempt to return 10 reflections & if the agglomeration returns less (and there are more than 10 other reflection groups in the meeting), then increase the grouping threshold iteratively until it works

However, if I change the grouping threshold from 0.25 to 0.95 or 0.05, I get no difference in results. The same number of reflections and reflection groups are returned. See [the issue here](https://github.com/ParabolInc/parabol/issues/5183) for more context and screenshots. Perhaps I'm missing something?

Reply:
what text are you using for the reflections?
the text has to be long enough & meaningful enough to actually work. it's gonna use a string similarity algo to detect how similar things are. if the reflection texts are too small, the similarity scores will be 0. might be worth using data from real retros to play with